### ISSUE TITLE
Add custom model selection

### ISSUE THREAD

**Issue by Neoony** [machinewrapped/gpt-subtrans#104]
Unless I missed it, I can only select the models you implemented and I cant enter a custom one.
Why not make it able for user to type any model he wants to use?

For example now new versions of models were released 
mainly gpt-4-0125-preview
but also gpt-3.5-turbo-0125

I never understand when someone makes this completely fixed/hardcoded :D
Less maintenance for you.

Please, Thanks!

By the way have to mention that this is absolutely amazing piece of software and I was hoping to find something like this for a good while (basically since GPT got popular)
And it also works amazingly well for me.
Big thanks for creating and working on it.

**Comment by machinewrapped** [comment]
The model list actually comes from the OpenAI integration library - I ask it for a list of supported models, and populate the drop-down with whatever it returns. It makes a request to OpenAI's servers to get the list, so if new models are not listed they're probably not accessible via the API yet - or perhaps it needs a library update to use them.

IIRC requesting a model that isn't in the supported list will just return a "model not supported" error, so being able to enter free text probably isn't going to help :-)

Thanks for the thanks!

**Comment by Neoony** [comment]
Ahh that makes sense now. If it gets it from the API then thats fine.

Actually I checked again and its in the list. (gpt-4-0125-preview - https://openai.com/blog/new-embedding-models-and-api-updates)
Was wondering if that was because I now tried your 0.5.3 from the other ticket
But rolled back to 0.5.1 and it seems to still be there

So either I actually missed it, or it actually wasnt getting it from the API when I checked 🤷‍♂️ (I guess that should not be, it was released a while ago, but I am never getting models from API myself in my GPT things, so cant tell )

My mistake I guess

...well this theme bug is not helping :D
![image](https://github.com/machinewrapped/gpt-subtrans/assets/24975075/e8110b60-b370-48e6-addb-72f44483f229)
Its on all the dropdowns in the dark theme (I definitely prefer dark themes)
